### PR TITLE
Proxy Settings for docker

### LINKED ISSUE #25: Support for proxy in docker daemon is missing
Hi, 

   If proxy needs to be configured for docker daemon, it is not supported in the daemon.json file. It has to be in the /etc/systemd/system/docker.service.d/http-proxy.conf. 
  
FYI https://docs.docker.com/engine/admin/systemd/#httphttps-proxy
Thanks

### PR DESCRIPTION
**Pull request by ageekymonk**
Fixes #25 by adding support for proxy

### CODE REVIEW

File: templates/docker.j2.service [hunk 1]
```diff
@@ -8,7 +8,7 @@ Type=notify
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/docker daemon -H unix:///var/run/docker.sock
+ExecStart=/usr/bin/dockerd -H unix:///var/run/docker.sock
```
Comment (mongrelion) [inline templates/docker.j2.service:11]: Was this change necessary? This has also come in issue #22 
If this fixes that, let's also close that one.

Comment (ageekymonk) [reply templates/docker.j2.service:11]: It is needed as the old way is deprecated. This fixes #22 as well

**Comment by mongrelion** [comment]
lgtm

**Comment by mongrelion** [comment]
This also fixes #22
